### DISCUSSION TITLE
[script] Sync folders with albums

### DISCUSSION THREAD

**Discussion by daniele-athome** [immich-app/immich#3382] [Community Guides]
Hello there,
While waiting for nested albums and/or a file system view or whatever will come out of the [Gallery Watcher](https://github.com/immich-app/immich/discussions/1006), I wrote a little script that will synchronize the folders in your (imported, not uploaded) library with albums having the folders name.

**This script is for people using exclusively imported libraries. I didn't test it with uploaded photos.**

The script: https://gist.github.com/daniele-athome/82ce8af869ec0e20cd73b69acbc589bd

The only dependencies (other than basic shell utils) are `curl` and `jq`.

Invocation:

```shell
./sync-albums.sh <http://immich-url/api> <api-key> <library-path>
```

`<library-path>` must be the same absolute path used as your imported library path. The path is only used to create album names; no access to the file system is needed.

### How it works

If you have a directory structure like:

```
/library/path/Misc
/library/path/World/New York
/library/path/People/Friends/Kind
```

Album names will be:

```
Misc
World/New York
People/Friends/Kind
```

To reflect the nature of a file system, the script will also make sure that any photo belongs to only one album (and it will remove it from all albums except the one that is supposed to belong to).

Everything is done through API so for huge libraries it will take some time. It's not bulletproof, but it works.

**Comment by Cantello** [comment]
Would it also be possible with this script to upload photos from existing folders into albums in Immich? I have a bunch of photos sorted in folders/albums (by year and sometimes occasion) and would like to have everything in a huge Immich database.

**Reply by daniele-athome** [reply]
It wouldn't work, because the script doesn't have a way to match the uploaded files with the original files (it works with import because the original filenames are preserved).

Anyway please note that this script doesn't import or upload anything: you run this script **after** you've imported your photos.

**Reply by Cantello** [reply]
Thanks for the explanation, I think I get it now and it might just be what I need (existing library with albums in folders).

**Comment by daniele-athome** [comment]
Many other alternatives also available in this discussion: https://github.com/immich-app/immich/discussions/4279
